drivers/net/dsa/lantiq_gsw_mdio.c

```diff
@@ -35,19 +36,103 @@
 #include "lantiq_gsw.h"
 #include "lantiq_pce.h"
 
-static u32 gsw_mdio_read(struct gswip_priv *priv, void *address)
+#define NUM_ACCESSIBLE_REGS (30)
+#define TARGET_BASE_ADDRESS_REG (31)
+
+struct gsw_mdio {
+	struct mdio_device *mdio_dev;
+	struct gswip_priv common;
+};
+
+static u32 gsw_mdio_read_actual(struct mdio_device *mdio, u32 reg)
```

Comment:
Probably can move the contents of this function into where it's called from. Same with the write `gsw_mdio_write_actual`